mypaint-brush.c

```diff
@@ -903,9 +918,12 @@ void print_inputs(MyPaintBrush *self, float* inputs)
     STATE(self, ACTUAL_ELLIPTICAL_DAB_ANGLE) = mod_arith(SETTING(self, ELLIPTICAL_DAB_ANGLE) - viewrotation + 180.0, 180.0) - 180.0;
   }
 
-  float *fetch_smudge_bucket(MyPaintBrush *self) {
+  SmudgeBucket *fetch_smudge_bucket(MyPaintBrush *self) {
     if (!self->smudge_buckets || !self->num_buckets) {
-      return &STATE(self, SMUDGE_RA);
+      float *smudge_state = &STATE(self, SMUDGE_RA);
+      // Not sure this can be done in general.
+      // static_assert (sizeof (SmudgeBucket) == sizeof(smudge_state) * SMUDGE_BUCKET_SIZE, "SmudgeBucket includes padding");
```

Comment:
Not sure this can be done in general. We might need something like `__attribute__((packed))`.